static/src/javascripts/projects/common/modules/component.js

```diff
@@ -107,7 +109,11 @@ define([
 
         this.elem = template;
         this._prerender();
-        bonzo(container)[this.manipulationType](this.elem);
+
+        fastdom.write(function () {
+            bonzo(container)[this.manipulationType](this.elem);
+        });
+
         this._ready();
```

Comment:
Possibly. Depends what the ready call does (but yeah, you're right, that sort of thing can cause race conditions). You could always do something to defer the execution of it inside the `fastdom.write` call if it's going to trigger off a whole bunch of things.